### PR TITLE
RavenDB-15776 - make sure that we don't exceed the maxMergeSize

### PR DESCRIPTION
**Pull request by grisha-kotler**

### CODE REVIEW

File: src/Lucene.Net/Index/LogMergePolicy.cs [hunk 1]
```diff
@@ -135,6 +138,17 @@ public virtual int MergeFactor
 	        }
 	    }
 
+        public virtual int NumberOfLargeSegmentsToMergeInASingleBatch
+		{
+            get { return numberOfLargeSegmentsToMergeInASingleBatch; }
+            set
+            {
+                if (value < 2)
+                    throw new System.ArgumentException("mergeFactor cannot be less than 2");
```
Comment (arekpalinski) [inline src/Lucene.Net/Index/LogMergePolicy.cs]: Wrong exception message - "mergeFactor"

File: src/Lucene.Net/Index/LogByteSizeMergePolicy.cs [hunk 2]
```diff
@@ -81,17 +83,31 @@ public virtual double MaxMergeMB
...
-	    /// the same level (even if they vary drastically in size)
-	    /// and will be merged whenever there are mergeFactor of
-	    /// them.  This effectively truncates the "long tail" of
-	    /// small segments that would otherwise be created into a
-	    /// single level.  If you set this too large, it could
-	    /// greatly increase the merging cost during indexing (if
-	    /// you flush many small segments). 
-	    /// </summary>
-	    public virtual double MinMergeMB
+        public virtual double LargeSegmentSizeMB
```
Comment (arekpalinski) [inline src/Lucene.Net/Index/LogByteSizeMergePolicy.cs]: Would be nice to put a comment here, explaining why do we actually want to have dedicated settings to _large_ segments.

File: src/Lucene.Net/Index/LogMergePolicy.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Lucene.Net/Index/LogMergePolicy.cs]: done

File: src/Lucene.Net/Index/LogByteSizeMergePolicy.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Lucene.Net/Index/LogByteSizeMergePolicy.cs]: done

**Pull request merged**